File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for(int num : nums){
            set.add(num);
        }

        int longestSequence = 0;
        for(int num:nums){
            if (!set.contains(num-1)){
                int currValue = num;
                int currLength = 1;
                while(set.contains(currValue+1)){
                    currValue+=1;
                    currLength +=1;
                }
                longestSequence = Math.max(longestSequence,currLength);
            }
        }
        return longestSequence;
    }
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) in [0,1]:
            return len(nums)
        # nums= list(set(nums))
        import heapq
        heapq.heapify(nums)
        curr_value = heapq.heappop(nums)
        count = 1
        ans = 1
        while True:
            try:
                next_value = heapq.heappop(nums)
            except Exception as e:
                    break
            if next_value-curr_value == 1:
                count += 1
                ans = max(count,ans)
            elif next_value-curr_value == 0:
                pass
            else:
                count = 1
                
            curr_value = next_value
        return ans
            
